src/main/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -212,7 +212,7 @@ protected VCFHeader parseHeaderFromLines( final List<String> headerStrings, fina
                     final VCFContigHeaderLine contig = new VCFContigHeaderLine(str.substring(9), version, VCFConstants.CONTIG_HEADER_START.substring(2), contigCounter++);
                     metaData.add(contig);
                 } else if ( str.startsWith(VCFConstants.ALT_HEADER_START) ) {
-                    final VCFSimpleHeaderLine alt = new VCFSimpleHeaderLine(str.substring(6), version, VCFConstants.ALT_HEADER_START.substring(2), Arrays.asList("ID", "Description"));
+                    final VCFSimpleHeaderLine alt = new VCFSimpleHeaderLine(str.substring(6), version, VCFConstants.ALT_HEADER_START.substring(2), Arrays.asList("ID", "Description"), Collections.emptyList());
```

Comment:
Well, its a bit ambiguous. The spec allows for all "structured" header lines to have optional fields, but you're right in that "Source" and "Version" are recommended specifically only for INFO. Technically, they should be accepted for ALT as well, but since we're still not implementing a fully generalized mechanism for optional attributes, it would probably be misleading to special case them for the other header lines.

Reply:
Okay, I see. I'll update the test file to exercise the valid options and make sure it works.
My understanding was that they want `expected` then `recommended` and only then `optional`, so technically if I parse ALT with Source and Version as `recommended`, I would disallow valid `##ALT<ID,Description,Random,Version,Random2,Source>`

Speaking of which, in my functions I have named Source/Version fields as "optional", while "recommended" is the more appropriate word. I will refactor so there is space for "truly optional".

Reply:
I think it will be sufficient to just make Source/Version work. Other attributes won't round trip since there is no place to store them, and thats a much bigger change, so we should try to keep this PR constrained to those.

Reply:
To my surprise, Source/Version seems to work and round-trip just fine. Even though the fields are not listed as recommended (technically they aren't), the same parsing code is used and if the fields are present, they are whisked into `source` and `version` members and round-tripped just fine. Since these values are not technically "recommended", I am not adding them to the `recommendedTags` list - but it still works fine.